AdvLoggerPkg/Application/DecodeUefiLog/DecodeUefiLog.py

```diff
@@ -598,22 +595,123 @@ def _GetLines(self, lines, LoggerInfo):
         CurrentLine = LoggerInfo["CurrentLine"]
         StartLine = LoggerInfo["StartLine"]
 
-        while (Status == self.SUCCESS):
-            (Status, MessageLine) = self._GetNextFormattedLine(MessageLine, LoggerInfo)
-            if Status != self.SUCCESS:
-                if Status != self.END_OF_FILE:
-                    print(f"Error {Status} from GetNextFormattedLine")
-                break
+        if LoggerInfo["Version"] == self.V4_LOGGER_INFO_VERSION:
+            # There is a potential that we may have enabled auto wrap.
+            # If so, we need to find the first legible line. Given that
+            # we read the logger information from the head of the buffer,
+            # we can start from this cursor as an acceptable estiamted
+            # start.
+            OriginalOffset = LoggerInfo["InFile"].tell()
+            StartOffset = LoggerInfo["LogCurrent"]
+            while True:
+                LoggerInfo["InFile"].seek(StartOffset)
+                CurrentOffset = StartOffset
+                ThisIsGood = True
+                # We need to verify that from this point on, we have at least
+                # one chunk of eligible entries. If not, we need to move along
+                # the pointer.
+                while CurrentOffset < LoggerInfo["LogBufferSize"] and\
+                        CurrentOffset < StartOffset + self.MAX_MESSAGE_SIZE:
+                    (MessageEntry, _) = self._ReadMessageEntry(LoggerInfo)
+                    CurrentOffset = LoggerInfo["InFile"].tell()
+                    if MessageEntry is None:
+                        ThisIsGood = False
+                        break
+
+                if ThisIsGood:
+                    LoggerInfo["InFile"].seek(StartOffset)

```

Comment:
This logic was somewhat confusing to me, can we add a comment here. This case I believe is we found a valid message after LogCurrent, so we are in the wrap case, right?

Reply:
This is saying that after the `LogCurrent`, we have found a starting point where the content has been good for at least an entire block (512 bytes). This is to prevent a memory writing across the block boundary during block readings.

We always assume we are in the wrap case and start to find the potential starting point. But if we have not identified a block that is fully formatted after `LogCurrent` till the end of the logging region, it means that we are not in the wrap case.

Reply:
I updated the flow as requested by Joey. Hopefully this is clearer now,